### DISCUSSION TITLE
new CT dataset

### DISCUSSION THREAD

**Discussion by d6222g** [voxelmorph/voxelmorph#603] [Q&A]
Hi im quit new to Voxelmorph and I do have a dataset (CT longitudinal data). I have already applied affine transformation and spacing is 1*1*1. the dataset dims are 512*512*n for each pair. how can i start training with this data? Any suggestions would be really appreciated!!!  Thank you

**Comment by adalca** [comment]
We have an extensive tutorial at https://tutorial.voxelmorph.net/, and train.py files. Do these help?

**Reply by d6222g** [reply]
@adalca Thank you very much for the quick response and really appreciate. I'm new to pytorch as well as 3D registration and these are my questions: 
1) After affine transformation with the voxel spacing [1 1 1] my fixed and moving images have different dimensions, so how can I add padding here and should i resize them to strictly 160 x 192 x 224 (but then the spacing gonna differ), how to deal with this?
2) How can I implement a custom generator for my 3D CT dataset - Im interested in pytorch version?
3) The list.txt can be /data/fixed/case_001.nii.gz, /data/moving/case_001.nii.gz (fixed and moving images in one line seperated by comma?)

Thank you very very much and really appreciate your support here!!!!!

**Reply by adalca** [reply]
1. if you are planning to train your own model, keep everything 512x512 (if your GPU allows), and initialize and train a model like that. Padding is not necessary then.
2. 3. It really depends on your data and what you want to do, but if you use a list.txt in the way you described in (3), then I think you can just use the scan to scan generator we have. I'll let @ahoopes chime in here since he wrote the torch scripts, but it should be that simple I think.

**Reply by d6222g** [reply]
Finally I have resampled my dataset to be 160x160x256 (/data/fixed and /data/moving) im planning to use the pytorch version rather than TF. Can someone help me with the next steps!!! Im bit confused because this is quit new to me.

**Reply by ghost** [reply]
> We have an extensive tutorial at https://tutorial.voxelmorph.net/, and train.py files. Do these help?

thanks for the tuto , really easy to understand . i can use the same code on CT data ?? is there other adjustments !!
